ecs-init/volumes/ecs_volume_plugin.go

```diff
@@ -70,6 +70,17 @@ func (a *AmazonECSVolumePlugin) LoadState() error {
 	if oldState.Volumes == nil {
 		return nil
 	}
+
+	// Reset volume mount reference count. This is for backwards-compatibility with old
+	// state file format which did not have reference counting of volume mounts.
+	for _, vol := range oldState.Volumes {
+		for mountId, count := range vol.Mounts {
+			if count == 0 {
```

Comment:
Q: Should this be `count <= 0` if we're also allowing the count to be negative? (based on L.44 of `ecs-init/volumes/types/types.go`)

Reply:
(please correct me if I'm misunderstanding this here)

Reply:
Currently the state file for the plugin looks like below 
```
            {
                "volumes": {
                    "efsVolume": {
                        "type":"efs",
                        "path":"/var/lib/ecs/volumes/efsVolume",
                        "options": {"device":"fs-123","o":"tls","type":"efs"},
                        "mounts": {"id1": null}
                    }
                }
            }
```

This code is to translate `"mounts": {"id1": null}` in old state files to `"mounts": {"id1": 1}`. JSON decoding a `null` as an `int` produces `0` by default and this code converts those `0`s to `1` to initialize existing mount IDs with a mount count of 1.